docs/source/API/core/parallel-dispatch/UniqueToken.rst

```diff
@@ -0,0 +1,100 @@
+``UniqueToken``
+===============
+
+Header File: ``Kokkos_Core.hpp``
+
+
+Description
+------------
+
+``UniqueToken`` is a portable way to acquire a unique ID for a thread (``thread-id`` is not portable across execution environments).  The unique ID is scoped by the ``UniqueTokenScope`` template parameter (defaults to ``Instance``, but can be set to ``Global``).
+
+Interface
+---------
+
+.. code-block:: cpp 
+
+    template <class ExecutionSpace = DefaultExecutionSpace, UniqueTokenScope = UniqueTokenScope::Instance> UniqueToken
+
+
+
+Parameters
+-----------
+
+*  ``ExecutionSpace``:  See `Execution Spaces <../execution_spaces.html>`_
+
+*  ``UniqueTokenScope``:  defaults to ``Instance``, and every instance is independent of another.  In contrast, ``Global`` uses one set of unique IDs for all instances.
+
+.. note::
+   In a parallel region, before the main computation, a pool of ``UniqueToken`` (integer) ID is generated.  A generated ID is released following iteration (see ``void release(size_t idx)`` below).
```

Comment:
I am not sure if we need the note when we have the warning below

Reply:
agreed

Reply:
I would prefer if we kept the warning and ditched the note. The warning was very explicit about what users have to adhere to. The note makes it sound optional in my opinion